### DISCUSSION TITLE
How to set window title text color

### DISCUSSION THREAD

**Discussion by resonant-jovian** [emilk/egui#929] [Q&A] [answered]
Hey!

Got a problem where the title bar color is not matching what I have set everywhere else:

![image](https://user-images.githubusercontent.com/23136737/144472866-d4809133-53cb-43b4-b08a-e8a2140e5976.png)

This is the code where I set the style:

```
        let mut style: egui::Style = (*ctx.style()).clone();

        style.visuals.extreme_bg_color = egui::Color32::from_rgb(45, 51, 59);

        style.visuals.faint_bg_color = egui::Color32::from_rgb(45, 51, 59);

        style.visuals.code_bg_color = egui::Color32::from_rgb(45, 51, 59);

        style.visuals.hyperlink_color = egui::Color32::from_rgb(255, 0, 0);

        style.visuals.override_text_color = Some(egui::Color32::from_rgb(173, 186, 199));

        style.visuals.window_corner_radius = 10.0;

        style.visuals.button_frame = true;

        style.visuals.collapsing_header_frame = true;

        style.visuals.widgets.noninteractive.bg_fill = egui::Color32::from_rgb(35, 39, 46);

        style.visuals.widgets.noninteractive.fg_stroke =
            egui::Stroke::new(0., egui::Color32::from_rgb(173, 186, 199));

        style.visuals.widgets.inactive.bg_fill = egui::Color32::TRANSPARENT;

        style.visuals.widgets.hovered.bg_fill = egui::Color32::from_rgb(45, 51, 59);

        style.visuals.widgets.active.bg_fill = egui::Color32::from_rgb(45, 51, 59);

        style.visuals.widgets.open.bg_fill = egui::Color32::from_rgb(45, 51, 59);

        ctx.set_style(style);

        let font_droidsansmono = include_bytes!("data/Droid Sans Mono Nerd Font Complete Mono.otf");
        let mut font = FontDefinitions::default();

        font.font_data.insert(
            "Droid Sans Mono".to_string(),
            Cow::from(&font_droidsansmono[..]),
        );
        font.fonts_for_family
            .insert(FontFamily::Monospace, vec!["Droid Sans Mono".to_string()]);

        font.fonts_for_family.insert(
            FontFamily::Proportional,
            vec!["Droid Sans Mono".to_string()],
        );
        /*
        font.family_and_size.insert(
            epaint::text::TextStyle::Body,
            (epaint::text::FontFamily::Proportional, 10.0),
        );
        font.family_and_size.insert(
            epaint::text::TextStyle::Body,
            (epaint::text::FontFamily::Monospace, 10.0),
        );
        */
        ctx.set_fonts(font); 
```

It is all in `fn setup`

/Albin

**Comment by coderedart** [comment]
not related, but that font is really great. good choice.

**Reply by resonant-jovian** [reply]
Droid sans mono for life

**Comment by emilk** [comment]
The title bar of what? The native window? That is outside what `egui` can control.

**Reply by resonant-jovian** [reply]
egui::window

**Reply by resonant-jovian** [reply]
In the example where it says Name

**Reply by emilk** [reply]
`style.visuals.widgets.noninteractive.bg_fill = egui::Color32::from_rgb(35, 39, 46);` controls that color. And it is what you set it to

**Reply by resonant-jovian** [reply]
Yeah but it appears a different colour than the others... Is there some Tone mapping going on?

**Reply by emilk** [reply]
`35, 39, 46` is the background color of the panels as well.

**Reply by resonant-jovian** [reply]
Yeah but there is are only three colours in the setup and the text Name is none of them

**Comment by emilk** [comment, marked as answer]
the text is in the foreground (`fg`), so it's `style.visuals.widgets.noninteractive.fg_stroke`. See https://docs.rs/egui/latest/egui/style/struct.WidgetVisuals.html

**Reply by resonant-jovian** [reply]
Will check that tomorrow

**Reply by resonant-jovian** [reply]
Works.

**Reply by Barugon** [reply]
> the text is in the foreground (`fg`), so it's `style.visuals.widgets.noninteractive.fg_stroke`. See https://docs.rs/egui/latest/egui/style/struct.WidgetVisuals.html

That changes the background color for the whole window. Is there any way to change the background color for just the title bar?

**Comment by delan** [comment]
This is the top result on Google for “egui title bar color”, so if anyone is wondering how to override the text color of a specific window: you can do it with [RichText](https://docs.rs/egui/0.23.0/egui/widget_text/struct.RichText.html), which is Into&lt;[WidgetText](https://docs.rs/egui/0.23.0/egui/widget_text/enum.WidgetText.html)> just like String:

```rust
Window::new(RichText::new("text").color(Color32::TEMPORARY_COLOR))
```
